### ISSUE TITLE
Using setVisible in beforeAddingWidget triggers an exception

### ISSUE THREAD

**Issue by hiker** [supertuxkart/stk-code#2453]
If setVisible is used in beforeAddingWidget, the state of the STK widget will be changed, but since the corresponding irrlicht element is not yet loaded, it can not be adjusted.

Example: online_profile_base.cpp sets the m_profile tabs to be invisible in beforeAddingWidget if the player is not logged in. Only later the irrlicht element is created, which still has the default visibility (true).

When later isVisible is tested, it detects the difference between the visibility state of the stk widget and irrlicht widget.

**Comment by hiker** [comment]
The originally found issue in online_profile_base was fixed in:
c6b22cdffb366f31295462a3a04fa33417c09ade

But we should have a look at other code. E.g. perhaps only allow changing visibility when an irrlicht element for an stk widget exists (though not sure if there might be some stk widgets that never have an irrlicht element), or at least print a warning. I've found at least one more place (some dialog - iirc when no server was found).
